### PR TITLE
Integrate metadata editor

### LINKED ISSUE #69: Integrate metadata editor
So after closing #45 I wrote https://github.com/space-nuko/sd-webui-model-metadata-editor and it works pretty well. However I'm finding a few downsides from having the functionality separated from additional_networks:

1. No ability to pass a model from additional_networks into the metadata editor and vice-versa. I have to go to the dropdown to select the model a second time even though it's already selected in one place. The dropdown is super inconvenient with hundreds of models and it's double the work each time
2. The metadata editor tab makes the additional networks tab redundant, the additional networks tab is pretty empty currently
3. Model hashing has to be done separately for each extension since they have no knowledge of each others' hashing caches, eating up additional time doing the same exact work two times. It makes much more sense for them to share a single cache
4. Less adoption of the metadata system since many more people use additional_networks than they do model-metadata-editor. If nobody knows the metadata is even there then it doesn't really help anyone

I do know it's a lot of code, but I'm mulling whether or not the extra maintenance burden is worth the tighter integration with additional_networks. Perhaps this integration could be accomplished while keeping the separation, for example with the "pass to txt2img" feature that could be done by exposing the Gradio components in additional_networks so they can be manipulated with event handlers in model-metadata-editor. (But in that case I'm not sure about the other way around, passing the models in additional_networks into model-metadata-editor)

### PR DESCRIPTION
**Pull request by space-nuko**
Closes #69

### CODE REVIEW

File: scripts/model_util.py [hunk 1]
```diff
@@ -0,0 +1,322 @@
...
+  extra_lora_paths = shared.opts.data.get("additional_networks_extra_lora_path", "").split(",")
+  for path in extra_lora_paths:
+    if os.path.isdir(path):
+      paths.append(path)
+
+  sort_by = shared.opts.data.get("additional_networks_sort_models_by", "name")
+  filter_by = shared.opts.data.get("additional_networks_model_name_filter", "")
+  res, res_legacy = get_all_models(paths, sort_by, filter_by)
+
+  lora_models = OrderedDict(**{"None": None}, **res)
```
Comment (kohya-ss) [inline scripts/model_util.py]: The ``Refresh models`` button in ``Additional Network`` on ``txt2img`` tab does not seem to work. I don't know why, but it seems to work if rewritten as follows Could you please verify?

```
  lora_models.clear()
  lora_models["None"] = None
  lora_models.update(res)
  # lora_models = OrderedDict(**{"None": None}, **res)
```

**Review by kohya-ss — COMMENTED** [review]
It looks quite good! Please verify the section I commented on.

**Comment by kohya-ss** [comment]
Thank you for PR! The metadata editor seems to be quite useful 😀

It would be nice to have documentation, even if it is brief. I will add it myself when I have time, but I would be happy to add it if you could.

**Comment by space-nuko** [comment]
I wrote a draft [here](https://github.com/space-nuko/sd-webui-additional-networks/blob/document-editor/README.md), it still needs a screenshot.

**Pull request merged**